### ISSUE TITLE
SSL Error when trying to install from test.pypi.org

### ISSUE THREAD

**Issue by kimberlythegeek** [pypa/python-packaging-user-guide#423]
Followed the instructions on [Uploading your Project to PyPi](https://packaging.python.org/tutorials/distributing-packages/#uploading-your-project-to-pypi), and [Using TestPyPi](https://packaging.python.org/guides/using-testpypi/#using-test-pypi).

I successfully uploaded my package to https://test.pypi.org/legacy/, and can view the latest release on both testpypi.python.org and test.pypi.org, but I keep receiving SSL errors when trying to install from the test site.

`$ pip install --index-url https://test.pypi.org/simple/ pypom-axe`

results in the following: 
```
Collecting pypom-axe
  Could not fetch URL https://test.pypi.org/simple/pypom-axe/: There was a problem confirming the ssl certificate: [SSL: TLSV1_ALERT_PROTOCOL_VERSION] tlsv1 alert protocol version (_ssl.c:661) - skipping
  Could not find a version that satisfies the requirement pypom-axe (from versions: )
No matching distribution found for pypom-axe
```

Previously (mid '17) the command I used to install from the test site was: `pip install -i https://testpypi.python.org/pypi` which worked for some time. 

Using this command with my current package results in the same error as above.

I can install from the production site with no problem.

___

Running MacOS High Sierra 10.13.2
Using `Python 2.7.14`, `pip 9.0.1`, `virtualenvwrapper==4.8.2`

Installed Python 2.7 via python.org.

Results  of `pip freeze` in the current venv:
```
asn1crypto==0.24.0
certifi==2017.11.5
cffi==1.11.2
chardet==3.0.4
cryptography==2.1.4
enum34==1.1.6
idna==2.6
ipaddress==1.0.19
pkginfo==1.4.1
pycparser==2.18
pyOpenSSL==17.5.0
requests==2.18.4
requests-toolbelt==0.8.0
six==1.11.0
tqdm==4.19.5
twine==1.9.1
urllib3==1.22
```

**Comment by jonparrott** [comment]
@dstufft any ideas here?

**Comment by di** [comment]
Hey @kimberlythegeek, could you tell us what version of OpenSSL you have? E.g.:

```
$ python -c "import ssl; print(ssl.OPENSSL_VERSION)"
OpenSSL 1.0.2m  2 Nov 2017
```

If you're using macOS and haven't upgraded OpenSSL already, you'll likely need to.

**Comment by kimberlythegeek** [comment]
`OpenSSL 0.9.8zh 14 Jan 2016`

I just updated to 1.0.2 using `brew install openssl`, but the output has not changed.

I added `export PATH="/usr/local/opt/openssl/bin:$PATH"` to my .zshrc file.

`openssl version` returns `OpenSSL 1.0.2n  7 Dec 2017`
`$ python -c "import ssl; print(ssl.OPENSSL_VERSION)"` returns `OpenSSL 0.9.8zh 14 Jan 2016`

**Comment by kimberlythegeek** [comment]
Evidently this is a problem with Mac and the Python 2.7 version distributed from python.org.

All of the solutions I found suggested to to install python via homebrew.

So, after much pain and suffering, I uninstalled python 2.7, removed symlinks, reinstalled from homebrew (which for some reason did not install pip), installed pip with get-pip.py, fixed references in my virtual environments, and finally it worked.

And there was much rejoicing. 🎉 

Thanks, guys.
